Felix Yuen iP

felfelyuen opened the pull request.

felfelyuen committed:
Level-0

felfelyuen committed:
Revert "Level-0"

This reverts commit 1b39d1cbd71a30dd999b7dc4e6a6ed2a4495d508.

felfelyuen committed:
Changed name of Helper

Changed name from Duke to Harry
Removed logo and added other greetings

felfelyuen committed:
Implemented Echo in main

Added Echo function to echo back inputs
Exits when "bye" is inputted

felfelyuen committed:
Implemented Task and TaskManager

Added Task class for each task
Added TaskManager class to edit tasks within array of tasks
Added PrintLine class to print a line

felfelyuen committed:
Added checkboxes and completed? functionality

Added isCompleted to each task
Checkboxes added to list now
Can mark or unmark boxes

felfelyuen committed:
Added Todo, Deadline, and Events

Added the aforementioned types between tasks
Edited the output messages for customisation

Stevexchia commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,21 @@
+        return task;
+    }
+
+    public boolean getCompleted() {
```
would be great if the boolean uses prefix so it can be followed if used in conditional statements :)

Stevexchia commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,45 @@
+    public void printTasks () {
+        PrintLine.printLine();
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < taskCounter; i++) {
+            System.out.print((i + 1) + ".");
+            printCompleted(tasks[i].getCompleted());
+            System.out.println(" " + tasks[i].getTask());
+        }
```
Good indentation & format, good job!

Stevexchia commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,45 @@
+        tasks[i].setCompleted(completed);
+        if (completed) {
+            PrintLine.printLine();
+            System.out.println("Nice! I've marked this task as done:");
+            System.out.println("[X] " + tasks[i].getTask());
```
Indentation for wrapped lines should be more spaces if I am not wrong!

Stevexchia commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,21 @@
+public class Task {
+    private String task;
+    private boolean isCompleted;
+
+    public Task(String task) {
```
Great usage of classes to provide code readability & understanding!

Stevexchia reviewed (commented):
Overall great job & progress so far, keep it up! 👍

HTY2003 commented on src/main/java/Harry.java:
```diff
@@ -0,0 +1,56 @@
+    }
+
+    public static void main(String[] args) {
+        TaskManager TaskList = new TaskManager();
```
Variable names should be in camelCase (e.g. TaskList should be taskList)

HTY2003 commented on src/main/java/PrintLine.java:
```diff
@@ -0,0 +1,8 @@
+public class PrintLine {
```
Class names are correctly written in PascalCase, good job!

HTY2003 replied:
However, class names should be written as nouns. This is particularly confusing as you have a class function with the same name as the class - PrintLine.printLine()

HTY2003 commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,45 @@
+    public void printTasks () {
+        PrintLine.printLine();
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < taskCounter; i++) {
```
You correctly used i as your iterator variable. Good job!

HTY2003 commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,45 @@
+public class TaskManager {
+    private Task[] tasks = new Task[100];
```
You used plural form on a collection of objects. Good job!

felfelyuen committed:
Added Todo, Deadline, Event classes, and updated code accordingly

Changed previous code to include aforementioned classes for better readability
Reformatted the main Harry code to handle input as an array of words

felfelyuen committed:
Edited code to fit coding standards

Made small edits to fit coding standards
Modified code with reference to comments

felfelyuen committed:
Added Brief Text-UI testing

Added very brief text UI testing, needs more comprehensive edits to include all functions

felfelyuen committed:
Added Exceptions

Added Exceptions and Error Handling to most functions (Add, Mark)

felfelyuen committed:
Finalised exceptions

Added additional exceptions that were missed out (empty list)

felfelyuen committed:
Merge branch 'branch-Level-5'

* branch-Level-5:
  Finalised exceptions
  Added Exceptions
  Added Brief Text-UI testing

felfelyuen committed:
Reformatted files to include packages

Created packages to include .harry.ui, .harry.exceptions and .harry.tasks
Edited code to include exception that was missed out

felfelyuen committed:
Merge branch 'branch-packages'

* branch-packages:
  Reformatted files to include packages

felfelyuen committed:
Changed tasks array to ArrayList

Changed format of tasks array into ArrayList, modified code to follow suit as well

felfelyuen committed:
Added Delete function

Added delete function
Changed some code for better readability
(eg: using printTask to replace some outdated code that have the same function)
Changed Exception name so that it is used for deletion as well

felfelyuen committed:
Implemented Save function

Added HandleFile for methods that handles the file
Edited Code within the task classes to include the boolean variable in constructor
Fixed bug to correctly format string input into task in addSpaces method

felfelyuen committed:
Handles Corrupted files

Added new exception for file reading functions to throw when file is corrupted

felfelyuen committed:
Merge branch 'branch-Level-6'

* branch-Level-6:
  Added Delete function
  Changed tasks array to ArrayList

felfelyuen committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Handles Corrupted files
  Implemented Save function

# Conflicts:
#	src/main/java/harry/ui/Harry.java
#	src/main/java/harry/ui/TaskManager.java

felfelyuen committed:
Made Code more OOP

Refactored code into more basic packages.
Added Packages (Printer, Parser, Storage)

felfelyuen committed:
Converted date format from String to LocalDateTime class

Converted date format and the corresponding methods.
Added exceptions handling for incorrect input handling

felfelyuen committed:
Added find function

Added printFoundTasks method to TaskManager
Added ProcessFinding method to Parser

felfelyuen committed:
Added header comments.

Added header comments to all classes and non-trivial methods in JavaDoc format.
Renamed some variables for clarity.

felfelyuen committed:
Accepts more formats of inputted date.

Added FormatDate method in HandleDate class to reformat dates of different formats.
Includes different more specific exceptions.

felfelyuen committed:
Merge pull request #1 from felfelyuen/branch-Level-8

Merge branch-Level-8

felfelyuen committed:
Merge branch 'master' into branch-Level-9

felfelyuen committed:
Merge branch 'branch-Level-9' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/harry/Parser/Parser.java
#	src/main/java/harry/Storage/HandleFile.java
#	src/main/java/harry/TaskList/Deadline.java
#	src/main/java/harry/TaskList/Event.java
#	src/main/java/harry/TaskList/Task.java
#	src/main/java/harry/TaskList/TaskManager.java

felfelyuen committed:
Merge pull request #2 from felfelyuen/branch-Level-9

Merge Branch-Level-9

felfelyuen committed:
Merge pull request #3 from felfelyuen/branch-A-JavaDoc

Merge-A-Java-Doc

felfelyuen committed:
Merge branch 'master' into branch-A-JavaDoc

Added some more JavaDoc comments.
Made edits for code quality improvement.

sevenseasofbri commented on src/main/java/harry/Exceptions/CorruptedFileException.java:
```diff
@@ -0,0 +1,4 @@
+package harry.Exceptions;
```
Package names should be in lower case. 
> From java coding standards: Names representing packages should be in all lower case.

sevenseasofbri commented on src/main/java/harry/Parser/Parser.java:
```diff
@@ -0,0 +1,142 @@
+package harry.Parser;
+
+import harry.Exceptions.*;
```
Avoid using wildcard imports. Import each class used explicitly.

sevenseasofbri commented on src/main/java/harry/Parser/Parser.java:
```diff
@@ -0,0 +1,142 @@
+            Printer.printError("the date seems off... type it in something like 'YYYY-MM-DD HH:MM' format");
+        } catch (InvalidDateException e ) {
+            Printer.printError("What is this lol, this ain't a valid date");
+        }
```
Good job maintaining the same level of abstraction throughout the method here

sevenseasofbri commented on src/main/java/harry/Parser/Parser.java:
```diff
@@ -0,0 +1,142 @@
+     * Determines what action the user wants to do and calls the relevant method.
+     * @param list TaskManager object that corresponds to list of tasks.
+     */
+    public static void processInput(TaskManager list) {
```
Line 87 and 101 have different spacing between the method name and the opening parentheses of the declaration. Probaby better to stick to one and maybe make it like line 101.

sevenseasofbri commented on src/main/java/harry/Parser/Parser.java:
```diff
@@ -0,0 +1,142 @@
+                processFinding(list, commands);
+                break;
+            case "mark":
+            case "unmark":
```
> From the java coding standard: The explicit `// Fallthrough` comment should be included whenever there is a case statement without a break statement.

sevenseasofbri commented on src/main/java/harry/Printer/Printer.java:
```diff
@@ -0,0 +1,54 @@
+ * Class of methods to print out messages.
+ * Includes printLine, printLogo, printStartingPage, printGoodbye, printError
+ */
+public class Printer {
```
Good use of OOP to declare a printing class.

sevenseasofbri commented on src/main/java/harry/TaskList/Deadline.java:
```diff
@@ -0,0 +1,33 @@
+        } else {
+            System.out.print("[ ] ");
+        }
+        System.out.print(taskName + " by: ");
```
Avoid the usage of magic literals by storing them in constants or vars

sevenseasofbri commented on src/main/java/harry/TaskList/Event.java:
```diff
@@ -0,0 +1,41 @@
+
+    public LocalDateTime getToDate() { return toDate; }
+
+    //public void setDate(LocalDateTime date) { this.date = date; }
```
Don't comment out dead code, you may just remove it

sevenseasofbri reviewed (commented):
Great job! Well written comments.

felfelyuen committed:
Update README.md

felfelyuen committed:
Update README.md

felfelyuen committed:
Update README.md

felfelyuen committed:
Updated README.md

felfelyuen committed:
Improved code quality.

felfelyuen committed:
Commented out unnecessary comments
